scalableminds/webknossos
# Add assertion for no duplicated keys in dataset and folder metadata in update routes

This PR adds an assertion to not have duplicate keys in metadata updates to datasets and folders.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- I suggest testing in firefox
- Open the dev instance, after it is loaded open the network tab
- Edit some metadata
- The network tab should show the update request. On this request use right click and select "edit and resent"
- Duplicate an already existing entry in the metadata of the dataset or folder and submit the request
- The request should result in a 400 bad request with the message: `Metadata keys must be unique.`

### Issues:
- fixes #8068

------
(Please delete unneeded items, merge only when none are left open)
